core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1468,6 +1510,28 @@ public Builder setPreviousFileLocation(String previousFileLocation) {
       return this;
     }
 
+    public Builder enableRowLineage() {
+      Preconditions.checkArgument(
+          formatVersion >= 3,
+          "Cannot use row lineage with format version %s. Only format version 3 or higher support row lineage",
+          formatVersion);
+      this.rowLineage = true;
+      changes.add(new MetadataUpdate.EnableRowLineage());
+      return this;
+    }
+
+    Builder incrementLastRowId(long delta) {
+      Preconditions.checkArgument(
```

Comment:
Maybe a test for this?

Reply:
changed this to private method, so now they only caller is addSnapshot, removed related test